File: 0101-0200/0119.pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        // C(k, i) = k! / (i ! * (k - i)!)
        // k! / (0! * k!), k! / (1! * (k-1)!)

        vector<int> ans;
        unsigned long long s = 1;
        ans.push_back(1);
        for (int i = 1; i <= rowIndex; ++i) {
            s = s * (rowIndex - i + 1) / i;
            ans.push_back((int)s);
        }
        return ans;
    }
};

File: 0101-0200/0119-pascals-triangle-ii.rs
struct Solution {}

impl Solution {
    pub fn get_row(row_index: i32) -> Vec<i32> {
        let k = row_index as usize + 1;
        let mut ans = vec![1; k];
        for i in 3..=k as usize {
            let mut last = 1;
            for j in 1..i - 1 {
                let y = ans[j] + last;
                last = ans[j];
                ans[j] = y;
            }
        }
        ans
    }
}

fn main() {
    for k in 0..=5 {
        println!("{:?}", Solution::get_row(k));
    }
}
